### ISSUE TITLE
Error occurs when I create subprocess after creating tf.train.Server with RDMA.

### ISSUE THREAD

**Issue by sleepfin** [tensorflow/tensorflow#20761]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Redhat 7.2
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: v1.8.0
- **Python version**: 2.7
- **Bazel version (if compiling from source)**: 0.11.0
- **GCC/Compiler version (if compiling from source)**: 4.8.5
- **CUDA/cuDNN version**: CUDA-9.0/cuDNN-7.0.5
- **GPU model and memory**: Tesla-P100
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

I compile the TensorFlow-v1.8.0 with RDMA support and modify the mnist_replicas.py based on: 
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/dist_test/python/mnist_replica.py

```
...
  if not FLAGS.existing_servers:
    # Not using existing servers. Create an in-process server.
    server = tf.train.Server(
        cluster, job_name=FLAGS.job_name, task_index=FLAGS.task_index, protocol='grpc+verbs')
    if FLAGS.job_name == "ps":
      server.join()
...
```

It works just fine.
Then I try to create a subprocess after `tf.train.Server` is created which leads to an error:

```
...
  if not FLAGS.existing_servers:
    # Not using existing servers. Create an in-process server.
    server = tf.train.Server(
        cluster, job_name=FLAGS.job_name, task_index=FLAGS.task_index, protocol='grpc+verbs')

    import multiprocessing; multiprocessing.Process(target=time.sleep, args=(100,)).start()

    if FLAGS.job_name == "ps":
      server.join()
...
```

 Here is the log:

worker:
```
...
job name = worker
task index = 0
2018-07-13 09:22:57.115547: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1356] Found device 0 with properties:
name: Tesla P100-PCIE-16GB major: 6 minor: 0 memoryClockRate(GHz): 1.3285
pciBusID: 0000:b5:00.0
totalMemory: 15.90GiB freeMemory: 15.27GiB
2018-07-13 09:22:57.115607: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1406] Ignoring visible gpu device (device: 0, name: Tesla P100-PCIE-16GB, pci bus id: 0000:b5:00.0, compute capability: 6.0) with Cuda compute capability 6.0. The minimum required Cuda capability is 7.0.
2018-07-13 09:22:57.115628: I tensorflow/core/common_runtime/gpu/gpu_device.cc:923] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-07-13 09:22:57.115636: I tensorflow/core/common_runtime/gpu/gpu_device.cc:929]      0
2018-07-13 09:22:57.115644: I tensorflow/core/common_runtime/gpu/gpu_device.cc:942] 0:   N
2018-07-13 09:22:57.116866: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job ps -> {0 -> 172.30.96.109:49998}
2018-07-13 09:22:57.116883: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job worker -> {0 -> localhost:49999}
2018-07-13 09:22:57.118450: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job ps -> {0 -> 172.30.96.109:49998}
2018-07-13 09:22:57.118468: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job worker -> {0 -> localhost:49999}
2018-07-13 09:22:57.125228: I tensorflow/contrib/verbs/rdma.cc:315] RoCE v2 is not configured for GID_INDEX 0
2018-07-13 09:22:57.127868: I tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc:332] Started server with target: grpc://localhost:49999
2018-07-13 09:22:57.134897: I tensorflow/contrib/verbs/rdma_mgr.cc:128] Connected to remote node /job:ps/replica:0/task:0
2018-07-13 09:22:57.770456: I tensorflow/contrib/verbs/rdma_mgr.cc:311] Instrumenting CPU allocator cpu_rdma_bfc
2018-07-13 09:22:57.770480: I tensorflow/contrib/verbs/rdma_mgr.cc:311] Instrumenting CPU allocator cpu_pool
WARNING:tensorflow:From mnist_replica_ib.py:217: __init__ (from tensorflow.python.training.supervisor) is deprecated and will be removed in a future version.
Instructions for updating:
Please switch to tf.train.MonitoredTrainingSession
Worker 0: Initializing session...
2018-07-13 09:22:58.064007: I tensorflow/core/distributed_runtime/master_session.cc:1136] Start master session 6cb605bf203f3160 with config: device_filters: "/job:ps" device_filters: "/job:worker/task:0" allow_soft_placement: true
2018-07-13 09:22:58.090607: F tensorflow/contrib/verbs/rdma.cc:691] Check failed: iter != request_table_.end()
Aborted
```

ps
```
...
job name = ps
task index = 0
2018-07-13 09:22:54.747787: E tensorflow/stream_executor/cuda/cuda_driver.cc:406] failed call to cuInit: CUDA_ERROR_NO_DEVICE
2018-07-13 09:22:54.747853: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:158] retrieving CUDA diagnostic information for host: bms-60c5
2018-07-13 09:22:54.747865: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:165] hostname: bms-60c5
2018-07-13 09:22:54.747923: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:189] libcuda reported version is: 390.67.0
2018-07-13 09:22:54.747988: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:193] kernel reported version is: 390.67.0
2018-07-13 09:22:54.748000: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:300] kernel version seems to match DSO: 390.67.0
2018-07-13 09:22:54.750241: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job ps -> {0 -> localhost:49998}
2018-07-13 09:22:54.750261: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job worker -> {0 -> 172.30.96.109:49999}
2018-07-13 09:22:54.751826: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job ps -> {0 -> localhost:49998}
2018-07-13 09:22:54.751843: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:215] Initialize GrpcChannelCache for job worker -> {0 -> 172.30.96.109:49999}
2018-07-13 09:22:54.759391: I tensorflow/contrib/verbs/rdma.cc:315] RoCE v2 is not configured for GID_INDEX 0
2018-07-13 09:22:54.762222: I tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc:332] Started server with target: grpc://localhost:49998
2018-07-13 09:22:57.770347: I tensorflow/contrib/verbs/rdma_mgr.cc:128] Connected to remote node /job:worker/replica:0/task:0
2018-07-13 09:22:57.770929: I tensorflow/contrib/verbs/rdma_mgr.cc:311] Instrumenting CPU allocator cpu_rdma_bfc
2018-07-13 09:22:57.770944: I tensorflow/contrib/verbs/rdma_mgr.cc:311] Instrumenting CPU allocator cpu_pool
```

Then I  try creating subprocess before `tf.train.Server` is created, it works without any error. (worker will stuck after finish training because subprocess is still sleeping.)
Code:
```
...
  if not FLAGS.existing_servers:
    # Not using existing servers. Create an in-process server.
    import multiprocessing; multiprocessing.Process(target=time.sleep, args=(100,)).start()
    server = tf.train.Server(
        cluster, job_name=FLAGS.job_name, task_index=FLAGS.task_index, protocol='grpc+verbs')
    if FLAGS.job_name == "ps":
      server.join()
...
```

Any idea?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by sleepfin** [comment]
@bignamehyp anyone help?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by byronyi** [comment]
@eladweiss Mind to take a look?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jeffdaily** [comment]
I've seen the same error before.  Try setting env var `RDMAV_FORK_SAFE=y`.  See https://www.rdmamojo.com/2012/05/24/ibv_fork_init/ for details.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by mohantym** [comment]
Hi @sleepfin! 1.x versions are not supported any more. You can use our [migration document ](https://www.tensorflow.org/guide/migrate)to use your 1.x code in 2.x version. Please create a new issue if it still persists in 2.7 version or later. Thank you!
